joblib/test/test_memory.py

```diff
@@ -569,8 +558,16 @@ def test_memory_in_memory_function_code_change(tmpdir):
 
 
 def test_clear_memory_with_none_cachedir():
-    mem = Memory(cachedir=None)
-    mem.clear()
+    memory = Memory(cachedir=None)
+    memory.clear()
+
+
+def func_with_kwonly_args(a, b, kw1='kw1', kw2='kw2'):
```

Comment:
I am guessing you added these functions only to get rid of some flake8 warnings. This is a bit misleading so I removed them. Basically the test is for python 3 only and defining a function whose name does not match its behaviour can just add confusion.